Added node.js support doc (#1346)
elastic/elasticsearch-js: docs/installation.asciidoc

To learn more about the supported major versions, please refer to the 
<<js-compatibility-matrix>>.

[discrete]
[[nodejs-support]]
=== Node.js support

NOTE: The minimum supported version of Node.js is `v8`.

> The client versioning follows the Elastic Stack versioning, this means that

Reviewer: ```suggestion
The client versioning follows the {stack} versioning, this means that
```

major, minor, and patch releases are done following a precise schedule that
often does not coincide with the https://nodejs.org/en/about/releases/[Node.js release] times.

> For avoiding supporting insecure and unsupported versions of Node.js, the

Reviewer: ```suggestion
To avoid support issues and unsupported versions of Node.js, the
```

Author: I want to keep the "insecure", as using an EOL version is not safe :)

client *will drop the support of EOL versions of Node.js between minor releases*.
Typically, as soon as a Node.js version goes into EOL, the client will continue
to support that version for at least another minor release. If you are using the client
with a version of Node.js that will be unsupported soon, you will see a warning
in your logs (the client will start logging the warning with two minors in advance).

> Because of the reasons defined above, unless you are *always* using a supported
> version of Node.js, we recommend defining the client dependency in your

Reviewer: ```suggestion
Unless you are *always* using a supported version of Node.js, 
we recommend defining the client dependency in your
```

`package.json` with the `~` instead of `^`. In this way, you will lock the

> dependency on the minor release and not the major. (eg: `~7.10.0` instead

Reviewer: ```suggestion
dependency on the minor release and not the major. (for example, `~7.10.0` instead
```
